qtpy/__init__.py

```diff
@@ -61,7 +61,7 @@
 import warnings
 
 # Version of QtPy
-from ._version import __version__
+__version__ = '2.0.0.dev0'
```

Comment:
I don't agree with this change because the previous way to set `__version__` is what we use in all our other projects. Its merits or flaws can be discussed in another PR if you want, but for now please leave things as they were, i.e.

```python
version_info = (2, 0, 0, 'dev0')
__version__ = '.'.join(map(str, version_info))
```

Reply:
I made every attempt to keep things exactly as they were here, but as `__version__` is not a static attribute and requires dynamic runtime code execution, `setuptools` fails to statically retrieve it from the AST during installation without importing the package and executing arbitrary code, which in turns triggers the complex, environment-dependent logic in `__init__.py` (which we _definitely_ don't want running on install) and crashes due to missing third party imports which haven't been installed yet. Furthermore, I tested defining a static `__version__` in `_version.py` and keeping everything else as it is, but it still triggers executing the `__init__` as before, and thus the same crash on install.

So, the simplest solution for this PR that didn't require risky hacks or much bigger changes (e.g. using `setuptools_scm`, a VERSION file, or another third-party tool) was to simply include `__version__` in `__init__.py` directly, particularly given the motivation of a separate `_version.py` to begin with was to allow the (now-legacy) setup.py to load and execute it, which is nowadays unnecessary and deprecated, and is exactly the behavior this PR migrates away from. See pypa/setuptools#1724 for more details (there were a few other workarounds there, but they require `__init__.py` to not have complex logic or third-party package imports, or rely on implementation-dependent hacks, and in any case didn't work in my testing).

Reply:
Ok, I see, but how does this work?

> a VERSION file

I mean, that sounds similar to what we had, doesn't it?

Reply:
Yeah, on the surface it may seem that way, but on closer investigation its actually quite different.

What we had before was an internal only (`_version.py`) Python module that both `__init__.py` could `import` and `setup.py` could `exec`.

The VERSION file I was referring to is a _plain text file_ (as typically used by third-party version management tools) with the version as a string, which `setup.cfg` would read in with `file:`, but for `__init__.py` to read it, we'd need to correctly include it in `package_data` and/or `MANIFEST.in` (with `include_package_data=True`) to ensure it was included in both sdists and wheels (or the package would crash on import), _and_ then open, read and parse it in `__init__,py` using `importlib.resources`/`importlib_resources`, which itself would either require unnecessarily dropping Python 3.6 support, or adding another third party dependency, increasing further the complexity and cost.

Given that we can now just include the `__version__` string right in `__init__.py` and `setup.cfg` based setuptools can simply read it directly from there with `attr:qtpy.__version__` without executing the file, without adding any of the complex, potentially risky gymnastics that the `file:` approach would require, this clearly seemed to be the simplest and most comparable working alternative.

Reply:
Hey @ccordoba12 this is the last open review comment on this PR; is this okay to merge now?

Reply:
i can't seem to find any use of `version_info` in the lib, but if it's of any importance, why not just doing it the other way around:

``` python
__version__ = "2.0.0.dev0"
version_info = __version__.split(".")
```

Reply:
Yep, thanks; I actually thought of that too in case any of our packages actually do use this value. In this case, it was never imported/exposed in `__init__` (or anywhere else public) to begin with, just in its own private internal-only module, so I didn't do that here. If for some reason we do want to, though, since its a constant in the top-level package namespace and wasn't here before, it would make sense to make it `VERSION_INFO`.

The issue with this (and equally, the old `version_info` -> `__version__` approach) versus just using `__version__` (which is what virtually all third party packages are likely to look for anyway, and has no real use internally) is that in the canonical PEP 440-normalized form, there isn't supposed to be a `.` between the last version component and alpha/beta/rc tags, only for  `devN`. This will get normalized in the canonical version in the package metadata, but is strongly discouraged and means the version stored in the package metadata, used by packaging tools and exposed via `importlib.metadata`, `pkg_resources`, etc will be inconsistent with the version in `__version__`. While this isn't a regression over the old approach, using `__version__` would free us from that limitation.

Reply:
> this clearly seemed to be the simplest and most comparable working alternative.

Hey @CAM-Gerlach, sorry for my late reply. Although I don't like too much, I think leaving things as they are right now is the best option.

Reply:
No worries, I know you have a lot going on and sorry for bugging you.

Okay, could you clarify what you mean by "leave things as they are"; do you mean leave things as they are in the PR or as they are in the repo (i.e. not merging the PR, or at least reverting the core improvements, and doing it over again later at some point)? Thanks!

Reply:
I mean leaving `__version__` in `__init__.py` as it's currently implemented in this PR.